compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1791,6 +1791,17 @@ impl<'a> Parser<'a> {
         Ok((fields, recovered))
     }
 
+    fn parse_unsafe_field(&mut self) -> Safety {
+        // not using parse_safety as that also accepts `safe`.
+        if self.eat_keyword(kw::Unsafe) {
+            let span = self.prev_token.uninterpolated_span();
```

Comment:
Why are you gating the uninterpolated span? Everywhere else in the parser seems to gate on `prev_token.span`.

Reply:
Probably there was no reason.